backend/x11/backend.c

```diff
@@ -231,20 +229,58 @@ static bool wlr_x11_backend_start(struct wlr_backend *backend) {
 		return false;
 	}
 
-	init_atom(x11, &x11->atoms.wm_protocols, 1, "WM_PROTOCOLS");
-	init_atom(x11, &x11->atoms.wm_delete_window, 0, "WM_DELETE_WINDOW");
-	init_atom(x11, &x11->atoms.net_wm_name, 1, "_NET_WM_NAME");
-	init_atom(x11, &x11->atoms.utf8_string, 0, "UTF8_STRING");
+	struct {
+		const char *name;
+		bool only_if_exists;
+		xcb_intern_atom_cookie_t cookie;
+		xcb_atom_t *atom;
+	} atom[] = {
+		{
+			.name = "WM_PROTOCOLS",
+			.only_if_exists = true,
+			.atom = &x11->atoms.wm_protocols,
+		},
+		{
+			.name = "WM_DELETE_WINDOW",
+			.only_if_exists = false,
+			.atom = &x11->atoms.wm_delete_window,
+		},
+		{
+			.name = "_NET_WM_NAME",
+			.only_if_exists = true,
+			.atom = &x11->atoms.net_wm_name,
+		},
+		{
+			.name = "UTF8_STRING",
+			.only_if_exists = false,
+			.atom = &x11->atoms.utf8_string,
+		},
+	};
+
+	for (size_t i = 0; i < sizeof(atom) / sizeof(atom[0]); ++i) {
+		atom[i].cookie = xcb_intern_atom(x11->xcb_conn,
+			atom[i].only_if_exists, strlen(atom[i].name),
+			atom[i].name);
+	}
+
+	for (size_t i = 0; i < sizeof(atom) / sizeof(atom[0]); ++i) {
+		xcb_intern_atom_reply_t *reply = xcb_intern_atom_reply(
+			x11->xcb_conn, atom[i].cookie, NULL);
+
+		if (reply) {
+			*atom[i].atom = reply->atom;
+			free(reply);
+		}
```

Comment:
I think this should have an `else *atom[i].atom = XCB_ATOM_NONE` (or does something else initialise these variables to 0?)

Reply:
We allocate most things with calloc, but I suppose that would be more obvious for people reading this code.

Reply:
This will be set to 0 by the way the struct is filled, non-specified fields in { .foo = bar, etc } are initialized to 0

Reply:
Sorry for double-replies, for some reason github didn't update with your comments when I logged in here.